### PR TITLE
Add seal shard api

### PR DESCRIPTION
**Pull request by sanebay**
Similar to create shard. 
Add test to create shard, seal and put blob and restart and verify shard is still sealed and put blob fails.

### CODE REVIEW

**Comment by zichanglai** [comment]
lgtm，it will be more clear once put/get blob pr is merged and rebase on that.

**Comment by sanebay** [comment]
> lgtm，it will be more clear once put/get blob pr is merged and rebase on that.

Rebased to latest. Please review.

File: src/lib/homestore_backend/tests/test_home_object.cpp
Comment (szmyd) [inline src/lib/homestore_backend/tests/test_home_object.cpp:334]: Are these different than the basic tests already in `lib/tests/ShardManagerTest.cpp` ? We should be using it for non-HS specific tests like these probably.

File: src/lib/homestore_backend/hs_shard_manager.cpp
Comment (szmyd) [inline src/lib/homestore_backend/hs_shard_manager.cpp:115]: We already called `get_shard` in `lib/shard_manager.cpp`; we should use its result not take the lock all over again. Probably means taking a `shard_it` rather than a `shard_id_t`.

File: src/lib/homestore_backend/hs_shard_manager.cpp
Comment (szmyd) [inline src/lib/homestore_backend/hs_shard_manager.cpp:149]: Can we just `return req->result()`?

File: src/lib/homestore_backend/hs_shard_manager.cpp
Comment (sanebay) [reply src/lib/homestore_backend/hs_shard_manager.cpp:115]: Will change the internal API to take a const ShardInfo&

File: src/lib/homestore_backend/hs_shard_manager.cpp
Comment (sanebay) [reply src/lib/homestore_backend/hs_shard_manager.cpp:149]: Yeah thats nice.

**Review by szmyd — APPROVED** [review]
lgtm

File: src/lib/homestore_backend/hs_shard_manager.cpp
Comment (yamingk) [inline src/lib/homestore_backend/hs_shard_manager.cpp:105]: Once a shard is sealed, the chunk backing this shard should be returned to heapChunkSelector, right?
Can you point me who is doing this part?

File: src/lib/homestore_backend/hs_shard_manager.cpp
Comment (yamingk) [inline src/lib/homestore_backend/hs_shard_manager.cpp:141]: We also need to update the shard's metablk from active to sealed state, right?

Comment (sanebay) [reply src/lib/homestore_backend/hs_shard_manager.cpp:141]: It was already there in hs_shard_manager.cpp on commit

File: src/lib/homestore_backend/hs_shard_manager.cpp
Comment (sanebay) [reply src/lib/homestore_backend/hs_shard_manager.cpp:105]: Yes this needs to be added.

**Review by zichanglai — APPROVED** [review]

**Pull request merged**
